plugin/lighthouse/handler.go

```diff
@@ -46,7 +46,10 @@ func (lh *Lighthouse) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns
 
 	var ips []string
 	var found bool
-	ip, found := lh.serviceImports.GetIP(pReq.namespace, pReq.service, pReq.cluster, lh.clusterStatus.IsConnected)
+
+	localClusterID := lh.clusterStatus.LocalClusterID()
+
+	ip, found := lh.serviceImports.GetIP(pReq.namespace, pReq.service, pReq.cluster, localClusterID, lh.clusterStatus.IsConnected)
```

Comment:
Won't it be simpler to just do something like this:

```
if pReq.cluster == "" {
    localClusterID := lh.clusterStatus.LocalClusterID()
}
ip, found := lh.serviceImports.GetIP(pReq.namespace, pReq.service, localClusterID,  lh.clusterStatus.IsConnected)
if !found {
    ip, found := lh.serviceImports.GetIP(pReq.namespace, pReq.service, pReq.cluster, lh.clusterStatus.IsConnected)
}```

With this we don't need to modify GetIP at all and reuse the existing cluster semantic.

Reply:
Hmmm I see what you mean, but first it will require 2 GetIP requests which require locks and are kind of expensive - won't you agree?
In addition, it will cause the logic IP selection to be kind of abstract away from where it is configured in the PR, and makes the selection kind of implicit and not explicit...
The logic that is within the map object explicitly says:
If a clusterID is provided in the DNS request, we would return the relevant IP for that cluster,  i.e if the service is not presented in that cluster we would like to return an empty IP.
While local cluster preference means - try to get the service IP from the local cluster, but if no presented fall back to remote once...

Reply:
Yeah, good to avoid multiple locks.